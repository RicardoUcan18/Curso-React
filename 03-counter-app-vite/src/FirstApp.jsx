import PropTypes from 'prop-types';

export const getResultado = () => { return 1 + 1 };

export const FirstApp = ({titulo, subtitulo}) => {

  return (
    <>
      <h1>{ titulo } </h1>
      {/* <h1>{ titulo } </h1> */}
      <p> { subtitulo }</p>
    </>
  )
}

FirstApp.propTypes = { 
  titulo: PropTypes.string.isRequired,
  subtitulo: PropTypes.number
}
FirstApp.defaultProps = {
  titulo: 'no funciona con defaultProps',
  subtitulo: 'subtitulo'
}


