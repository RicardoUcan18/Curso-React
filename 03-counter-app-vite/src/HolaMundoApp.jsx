export function App() {
  return (<h1> Hola mundo1!!! </h1>);
}