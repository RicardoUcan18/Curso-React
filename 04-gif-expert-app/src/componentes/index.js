export * from './AgregaCategoria';
export * from './GifGrid';
export * from './GifItem';