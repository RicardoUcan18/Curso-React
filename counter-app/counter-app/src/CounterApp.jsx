import { useState } from "react";
import PropTypes from "prop-types";


export function CounterApp({ value }) {

    const [counter, setCounter] = useState( value );

    const handleClick = ()  =>{
      setCounter( counter + 1 );
    }

    const handleClickminus = ()  =>{
      setCounter( counter - 1 );
    }
    
    
      return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ handleClick }> 
                +1 
            </button>

            <button onClick={ handleClickminus }> 
                -1 
            </button>

            <button onClick={ () => setCounter( value ) }>
                Reset
            </button>


        </>
      );
}

//validacion de las propiedades del componente
CounterApp.propTypes = {
    value: PropTypes.number.isRequired, 
};
