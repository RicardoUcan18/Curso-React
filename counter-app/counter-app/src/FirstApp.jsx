import {Fragment} from 'react';

import PropTypes from 'prop-types';

const newMessage = 'Hola, soy un mensaje nuevo';
const newMessage2 = [1,2,3,4,5,6,7,8,9,10];
const newMessage3 = {
    mensaje: 'Hola, soy Alvaro',
    titulo : 'Nombre',
};

const getResult = () => {
    return 'Hola, soy un mensaje desde una funcion';
}

export function FirstApp({ title, subtitle }) {
    
      return (
        <>
          {/* <h1>{ JSON.stringify(newMessage3) }</h1> */}
          <h1>{ title }</h1>
          <p>{ subtitle }</p>
        </>
      );
}

//validacion de las propiedades del componente
FirstApp.propTypes = {
    title: PropTypes.string.isRequired, 
};

//propiedades por defecto
FirstApp.defaultProps = {
    subtitle: 'No hay subtitulo',
    title: 'No hay titulo',
};

