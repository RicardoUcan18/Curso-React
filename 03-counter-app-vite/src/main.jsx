import React from 'react';
import ReactDOM from 'react-dom/client';
import { App } from './HolaMundoApp';
import { FirstApp } from './FirstApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={1} />
    {/* <FirstApp titulo='Mi primera' subtitulo='appa'/> */}
  </React.StrictMode>
);