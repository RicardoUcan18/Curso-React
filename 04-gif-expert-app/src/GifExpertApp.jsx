import { useState } from 'react'
import { AgregaCategoria, GifGrid} from './componentes';
// 
export const GifExpertApp = () => {

  const [categorias, setCategorias] = useState(['One puch'])
  const onAgregaCategoria = (iCategoria) => {
    if (categorias.includes(iCategoria)) return;

    setCategorias([iCategoria, ...categorias]);
  }
  return (
    <>
      <h1>Gif Expert App</h1>
      <AgregaCategoria
        onAgregaNuevaCategoria={onAgregaCategoria}
      />
      {
        categorias.map(
          categoria => (
            <GifGrid key={categoria} categoria={categoria} />
          )
        )
      }
    </>
  )
}
