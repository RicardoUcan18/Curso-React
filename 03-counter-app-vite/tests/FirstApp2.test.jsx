import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas FirstApp', () => { 
  const titulo = 'titulo desde prueba';

  test('debe ser match con el snapshot ', () => { 
    const {container} = render(<FirstApp titulo={titulo}/>);
    expect(container).toMatchSnapshot();
   });

   test('debe de mostrar el titulo en h1', () => { 
    
     render(<FirstApp titulo={titulo}/>);
     expect(screen.getByText(titulo)).toBeTruthy();
    });
    test('debe de mostrar el titulo en un H1', () => { 
      
     render(<FirstApp titulo={titulo}/>);
      expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(titulo);
     })
 })