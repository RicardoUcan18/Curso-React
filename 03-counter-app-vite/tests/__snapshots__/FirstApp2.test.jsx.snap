// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas FirstApp debe ser match con el snapshot  1`] = `
<div>
  <h1>
    titulo desde prueba
     
  </h1>
  <p>
     
  </p>
</div>
`;
